webapp/src/immersivereader.tsx

```diff
@@ -46,29 +60,98 @@ function beautifyText(content: string): string {
-            /\*\*([^*]+)\*\*/gu,
-            (matched, word, offset, s) => lf("\"{0}\"", word)
+            /^```block$[^`]*^```$/gum,
+            ""
+        );
+    }
+
+    // Remove ** and __ around bold text, replace with " if it's only one word
+    function cleanBold(content: string): string {
+
+        let singleWordBold = content.replace(
+            /\*\*([^*^\s]+)\*\*|__([^\s^_]+)__/gu,
+            (matched, boldStar, boldUnder, offset, s) => {
+                return "\"" + lf("{0}", (boldStar ? boldStar : boldUnder)) + "\""
+            }
+        )
+
+        return singleWordBold.replace(
+            /\*\*([^*]+)\*\*|__([^_]+)__/gu,
+            (matched, boldStar, boldUnder, offset, s) => {
+                return boldStar ? lf("{0}", boldStar) : lf ("{0}", boldUnder);
+            }
         );
     }
 
     // replace horizontal rules with new lines
     function cleanHorizontalRule(content: string): string {
         return content.replace(
-            /^\* \* \*$/gum,
-            "- - -"
+            /^\* \* \*$|^- - -$|^<hr\/>$/gum,
+            "<br /><br />"
         );
     }
 
     // replace consecutive new lines with break tags so that they don't get smooshed
     function cleanNewLines(content: string): string {
         return content.replace(
             /[\r\n]{2,}/gum,
-            `<br />`
+            `<br />\n`
         );
     }
+
+    // Remove >> for blockquotes
+    function cleanBlockquotes(content: string): string {
+        return content.replace(
+            /^>>/gu,
+            ""
+        )
+    }
+
+    // Remove all emojis because they get read out. Characters that aren't
+    // emojis don't get hit by this RegEx, but they're silent
+    function cleanEmojis(content: string): string {
+        let PICTOGRAPHIC_REGEX: RegExp;
+        try { // Some browsers do not support unicode property escape, in which case we can just use _ replacement
+            PICTOGRAPHIC_REGEX = new RegExp("\\p{Extended_Pictographic}", "ug")
+        } catch {}
+        const specialEmojis: string[] = [];
+        if (PICTOGRAPHIC_REGEX) {
+            return content.replace(
+                PICTOGRAPHIC_REGEX,
+                (emoji, capture, offset, s) => {
+                    if (emoji == "🔲") {
+                        return "•"
+                    } else if (specialEmojis.indexOf(emoji) < 0) {
+                        return ""
+                    } else {
+                        return emoji
+                    }
+                }
+            )
+        }
+        return content;
+    }
+
+    function cleanItalics(content: string): string {
+        return content.replace(
+            /\*([^*]+)\*|_([^_]+)_/gu,
+            (matched, starItalics, underlineItalics, offset, s) =>
+                lf("{0}", starItalics ? starItalics : underlineItalics)
+        );
+    }
+
+    function convertBoolAngles(content: string): string {
+        return content.replace(
+            /<(true|false)>/gu,
```

Comment:
small: do we also want this for the other toggles (win/lose, on/off etc)? there's a pretty good list of them here https://github.com/microsoft/pxt/blob/master/libs/pxt-common/pxt-helpers.ts#L588, but i tried it on some platformer tutorials with the win/lose and it seemed fine with the angle brackets also

Reply:
!!! yes we definitely want for other toggles tytyty Immersive Reader ignores everything inside a <> so it sounds silly when it takes it away :(